teacher.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require './person'
+
+# Represents a teacher with a specific specialization.
+class Teacher < Person
+  def initialize(specialization)
+    super
+    @specialization = specialization
+  end
```

Comment:
Just as I suggested above, the same should apply here😉.  but in this case, the constructor extends the parent's constructor by adding @specialization and a parameter for it😉. You can take a look at the suggestion I gave in the previous comment to have an idea of how it can be implemented here😉.